core/src/main/scala/com/mesosphere/usi/core/SchedulerLogicGraph.scala

```diff
@@ -42,7 +43,7 @@ object SchedulerLogicGraph {
   * It's existence is only warranted by forecasted future needs. It's kept as a graph with an internal buffer as we will
   * likely need timers, other callbacks, and additional output ports (such as an offer event stream?).
   */
-private[core] class SchedulerLogicGraph(mesosCallFactory: MesosCalls, initialState: StateSnapshot)
+private[core] class SchedulerLogicGraph(mesosCallFactory: MesosCalls, initialState: StateSnapshot, metrics: Metrics)
```

Comment:
I wonder whether it's time for a dependency injection library. What do you think @timcharper and @ANeumann82?

Reply:
Hmm, we could perhaps consider something akin to ActorSystem which holds the configuration for USI and a lot of the environment specific configuration.

Reply:
Hmmmm.. Might be. I'd have to look deeper into the whole structure. If we're going with DI, we need to make sure it's pretty encapsulated, as we're providing a library here, not a full app... I think for now it's fine without